### DISCUSSION TITLE
[@types/auth0] popUpOrigin option not supported in the typescript implementation

### DISCUSSION THREAD

**Discussion by amezian92** [DefinitelyTyped/DefinitelyTyped#54934] [Issues with a @types package]
I have a question regarding the DefinitelyTyped Auth0 library, I’m currently using Types Auth0-JS and its working like a charm, but recently I had a requirement to change the redirection URL (Callback URL) in Auth0 Popup Mode, and the new URL domain is different from the original domain which caused the below error: 

> “error:"origin_mismatch", The redirectUri's origin test.com should match the window's origin example.com.”

 

The flow that I am trying to achieve is:

	

- The Library is initialized on example.com domain

- The redirection URL given is test.com/callBackUri.

- The test.com/callBackUri page resolves the results and closes the Popup.

After searching the error and checking Auth0 possible solutions, I found that there is an option to bypass the origin mismatch which is popupOrigin option, so I tried to register the test.com domain in Auth0 CORS configurations and found that in the Auth0 JS library they have the below code snippet that handles the popUpOrigin Option:

 

```
WinChan.onOpen(function(popupOrigin, r, cb) {

if (popupOrigin !== originUrl) {

 return cb({

 error: 'origin_mismatch',

 error_description:

 "The popup's origin (" +

 popupOrigin +

 ') should match the `popupOrigin` parameter (' +

 originUrl +

 ').'

 });

 }

_this.webAuth.parseHash(options||{}, function(err, data) {

 return cb(err || data);

 });

 });

};

```
In the Types Auth0-JS implementation, the following code is triggered which enforces the CORS check, even though I sent the popUpOrigin option when initializing the web auth in my typescript project.

 
```

var currentOrigin=windowHelper.getOrigin(),

redirectUriOrigin=objectHelper.getOriginFromUrl(options.redirectUri);

if(redirectUriOrigin&&currentOrigin!==redirectUriOrigin)

return cb({error:"origin_mismatch",error_description:"The redirectUri's origin ("+redirectUriOrigin+") should match the window's origin ("+currentOrigin+")."});
```

 

Is there a solution in the Types Auth0-JS library that I can use to overcome this error and achieve my goal to use different domains?

**Comment by amezian92** [comment]
any help would be appriciated

**Reply by peterblazejewicz** [reply]
what is the option in Auth0 docs you're referring to? thx!

**Reply by amezian92** [reply]
Hello @peterblazejewicz please find the option in the following file 
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/auth0-js/index.d.ts
`export interface AuthOptions {
 ...
    **popupOrigin?: string | undefined;**
...
} `

[a comment by an automated account is not shown]
